### ISSUE TITLE
[Feature Request] Steam currently running audio stream

### ISSUE THREAD

**Issue by aquac** [wdehoog/donnie#13]
Would it be possible to implement a feature like https://github.com/masmu/pulseaudio-dlna, that allows to stream currently running audio stream to the receiver? 
Application case is e.g. that I have cutespot (spotify) running on my phone and want to stream it to the receiver. Or I found some podcast while browsing with the phone and instead of download (what is sometimes difficult from the phone depending on the platform where it is published) copying it to the content server would be a significant overhead or require a laptop or desktop PC.

**Comment by wdehoog** [comment]
As a control point donnie can request a remote Renderer to play a stream (url). It does not send any audio by itself. Currently the streams it knows about are to be provided by a Content Server. If I understand you correctly you want to have the option to send a request to play a particular url to the Renderer? Technically it should be possible but how would this be implemented in the UI?

The project you refer to seems to 'cast' the pulse audio output to a remote server. This is some layers below where donnie acts. I think this needs an app on its own that also allows to control pulse audio sinks and source on the fly.

**Comment by aquac** [comment]
Thanks for your reply!

Would it be possible to send links via "share link" from within the browser? Or do these link that can be played by the renderer need particular formatting?

**Comment by wdehoog** [comment]
No. "share link" from within the browser requires an account plugin which seems overkill.

I added a "Add Stream" to the pully menu on the player and renderer pages. You can enter the url and a title. So you can for example paste a link copied from the browser. 

This feature is not present in the openrepos version. If you want to give it a try you can install it from my OBS repository (see readme.md).

I also added mpris openUri support (a DBus interface specification). Maybe this can be used from other apps, maybe even the browser, but I don't know that. (I am using it in my shoutcast app.)
